React/Base/RCTComponentEvent.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <React/RCTEventDispatcher.h>
+
+/**
+ * Event that is dispatched to React components.
```

Comment:
Should we discribe here that it is general purpose event? Should/can it be subclasses?

Reply:
It could be subclassed but just reimplementing RCTEvent is probably better especially now that a bunch of methods are optional.